test/NuGet.Core.FuncTests/Dotnet.Integration.Test/ProjectFileBuilder.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        public void Build(MsbuildIntegrationTestFixture fixture, string path)
+        {
+            BaseDir = path;
+
+            fixture.CreateDotnetNewProject(path, ProjectName, " classlib");
+
+            ModifyProjectFile();
+        }
+
+        /// <summary>
+        /// Adds and intem inside a &lt;ItemGroup/&gt; node in the following form:
+        /// <c>&lt;{itemType} Include="{itemPath}" [PackagePath="{packagePath}"] /&gt;</c>
+        /// </summary>
+        public ProjectFileBuilder WithItem(string itemType, string itemPath, string packagePath)
+        {
+            ItemGroupEntries.Add(new ItemEntry 
+            {
+                ItemType = itemType, 
+                ItemPath = itemPath, 
+                PackagePath = packagePath,
+            });
+
+            return this;
+        }
+
+        public ProjectFileBuilder WithPackageIcon(string packageIcon)
+        {
+            PackageIcon = packageIcon;
+
+            return this;
+        }
+
+        public ProjectFileBuilder WithPackageIconUrl(string packageIconUrl)
+        {
+            PackageIconUrl = packageIconUrl;
+
+            return this;
+        }
+
+        public ProjectFileBuilder WithProjectName(string projectName)
+        {
+            ProjectName = projectName;
+
+            return this;
+        }
+
+        public ProjectFileBuilder WithProperty(string propertyName, string value)
+        {
+            Properties[propertyName] = value;
+
+            return this;
+        }
+
+        private void ModifyProjectFile()
+        {
+            using (FileStream stream = new FileStream(ProjectFilePath, FileMode.Open, FileAccess.ReadWrite))
+            {
+                var xml = XDocument.Load(stream);
+
+                if (PackageIconUrl != null)
+                {
+                    ProjectFileUtils.AddProperty(xml, "PackageIconUrl", PackageIconUrl);
+                }
+
+                if (PackageIcon != null)
+                {
+                    ProjectFileUtils.AddProperty(xml, "PackageIcon", PackageIcon);
+                }
+
+                ProjectFileUtils.AddProperties(xml, Properties);
+
+                var attributes = new Dictionary<string, string>();
+                var properties = new Dictionary<string, string>();
+                attributes["Pack"] = "true";
+                foreach (var tup in ItemGroupEntries)
+                {
+                    attributes.Remove("PackagePath");
+
+                    if (tup.PackagePath != null)
+                    {
+                        attributes["PackagePath"] = tup.PackagePath;
+                    }
+
+                    ProjectFileUtils.AddItem(xml, tup.ItemType, tup.ItemPath, string.Empty, properties, attributes);
+                }
+
+                ProjectFileUtils.WriteXmlToFile(xml, stream);
+            }
+        }
+    }
+
+    /// Represents an MsBuild project file item entry,
+    /// For testing purposes
+    public class ItemEntry
```

Comment:
```suggestion
    internal class ItemEntry
```

Reply:
but this should also be a nested class, or in a separate file (although it's only used inside this other class in this file, so it should be nested)
Also, if the `///` comments are supposed to be XMLdoc, you're missing the XML. Otherwise just use `//` normal comments.

Reply:
ok, will change that